### PR TITLE
Updating Roslyn to 2.7.0-beta3-62529-05.

### PR DESCRIPTION
**Pull request by livarcocc**

### CODE REVIEW

**Comment by livarcocc** [comment]
@jaredpar this is for 15.6.

**Comment by dasMulli** [comment]
Is there any danger of beta 7.3 language features lighting up for LangVersion=Latest projects only in the 15.6 CLI but not in 15.6 VS with this 2.7 package?

**Comment by jaredpar** [comment]
@dasMulli 

> Is there any danger of beta 7.3 language features lighting up for LangVersion=Latest projects only in the 15.6 CLI but not in 15.6 VS with this 2.7 package?

No. C# 7.3 isn't included in 15.6, or this package, hence won't show up due to this package. 

The more general question though:

> Is there any danger of language features working from a CLI build but having squiggles inside VS

Yes that can happen. If you have a CLI that supports a language newer than the one in VS then you can get into this error. Thats true for both Latest and explicitly set versions.

**Comment by dasMulli** [comment]
Thanks, I think I just got confused by roslyn version numbers. (just seen that there are 2.8 packages now as well).

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

File: src/redist/redist.csproj [hunk 1]
```diff
@@ -84,6 +84,7 @@
       <RoslynDeps Include="$(RoslynDirectory)/bincore/csc.deps.json" />
       <RoslynDeps Include="$(RoslynDirectory)/bincore/vbc.deps.json" />
       <RoslynSatelliteDeps Include="$(RoslynDirectory)/bincore/tool_roslyn_satellites.deps.json" />
+      <RoslynFrameworkAssemblies Include="$(RoslynDirectory)/System.*.dll;$(RoslynDirectory)/runtimes/**/System.*.dll" Exclude="$(RoslynDirectory)/runtimes/**/System.IO.Pipes.AccessControl.dll"/>
```
Comment (nguerrera) [inline src/redist/redist.csproj]: What's going on here?

Comment (nguerrera) [reply src/redist/redist.csproj]: How did binaries in shared framework end up in roslyn package?

Comment (jaredpar) [reply src/redist/redist.csproj]: We had to add a reference to System.IO.Pipes.AccessControl in order to make the compiler server work. Once we did that the SDK started dumping the extra assemblies into our output directory. Essentially this is what SDK thought we should ship hence we shipped it.

Comment (nguerrera) [reply src/redist/redist.csproj]: What version of System.IO.Pipes.AccessControl are you referencing and what version of .NETCoreApp are you publishing for?

Comment (nguerrera) [reply src/redist/redist.csproj]: I need to understand what prompted this change. From past conversations, we fully expect Roslyn to carry new-than-shared-framework versions of System.Collections.Immutable and System.Reflection.Metadata in the near future. It feels wrong to delete stuff out from under what Roslyn published.

**Comment by nguerrera** [comment]
@MattGertz for approval of Roslyn insertion in to CLI for 15.6 P4. I will hold on merging until I understand the issue with framework assemblies, though.

**Comment by MattGertz** [comment]
Proactively approving.

File: src/redist/redist.csproj (hunk 1 above)
Comment (nguerrera) [reply src/redist/redist.csproj]: Learned offline that deleting these assemblies is due to crossgen failing. I'm going to pull down the change and see if there's a less fragile fix.

Comment (nguerrera) [reply src/redist/redist.csproj]: I see that the problem is that the build task is published for netstandard1.3, which is why it thinks all these assemblies are needed. Multi-targeting the build task (like sdk and nuget do) would get us out of the business of deleting these files. I will file a bug on Roslyn to do that.

**Comment by livarcocc** [comment]
@nguerrera are you satisfied with this change now? Have you filed an issue against Roslyn? I intended to, but then the flu got me.

**Comment by nguerrera** [comment]
Haven't filed the issue. I will do so. I am satisfied if we're tracking it as a temporary workaround against a roslyn bug.

**Comment by nguerrera** [comment]
Filed https://github.com/dotnet/roslyn/issues/24646

File: src/redist/redist.csproj (hunk 1 above)
Comment (nguerrera) [reply src/redist/redist.csproj]: https://github.com/dotnet/roslyn/issues/24646

**Pull request merged**